drivers/led/ht16k33.c

```diff
@@ -230,15 +229,13 @@ static bool ht16k33_process_keyscan_data(const struct device *dev)
 			pressed = true;
 		}
 
-		if (data->kscan_cb == NULL) {
-			continue;
-		}
-
 		for (col = 0; col < HT16K33_KEYSCAN_COLS; col++) {
-			if (changed & BIT(col)) {
-				data->kscan_cb(data->child, row, col,
-					state & BIT(col));
+			if ((changed & BIT(col)) == 0) {
+				continue;
 			}
+			input_report_abs(dev, INPUT_ABS_X, col, false, K_FOREVER);
+			input_report_abs(dev, INPUT_ABS_Y, row, false, K_FOREVER);
+			input_report_key(dev, INPUT_BTN_TOUCH, state & BIT(col), true, K_FOREVER);
```

Comment:
Not sure this is correct? The HT16K33 is not a touch screen controller. It's a keyboard matrix driver. Why report X/Y coordinates and touch events? What am I missing?

The existing callback solution predates the input subsystem, forcing row/col decoding to be done at the application level.

Reply:
Yeah but I ended up keeping the x/y/touch convention for keyboard matrix drivers because there are known use cases where you want the row/column straight from the driver (plus it allows backward compatibility), so by doing this you can:
- use the kscan compatibility driver
- use whatever code you want to process row/col/touch (for example feeding them on a layering system, think zmk, or send them straight to the AP, we have that in chromebooks)
- use a `input-keymap` subdevice to translate the output into `INPUT_KEY_` events
- any combination of the above really

Reply:
Thank you for that explanation, that makes sense.